spec/lib/guard/notifier_spec.rb

```diff
@@ -41,4 +41,42 @@
       end
     end
   end
+
+  describe "backward compatibility" do
+    let(:notifier) { instance_double("Notiffany::Notifier") }
+
+    before do
+      allow(Notiffany::Notifier).to receive(:new).and_return(notifier)
+    end
+
+    after do
+      Guard::Notifier.instance_variable_set(:@notifier, nil)
+    end
+
+    context "multiple parameters" do
+      before do
+        subject.connect
+        allow(notifier).to receive(:notify)
+          .with("A", { priority: 2, image: :failed }).and_return(true)
```

Comment:
Since we're testing notification here, we'd prefer to stub all invocations, so just having `allow(notifier).to receive(:notify)` alone is enough (tests will place expectations on actual parameters anyway - so no need to duplicate the example parameters).